kedro_vertexai/generator.py

```diff
@@ -220,3 +224,81 @@ def _configure_resources(self, name: str, operator):
         if "memory" in resources and resources["memory"]:
             operator.set_memory_limit(resources["memory"])
             operator.set_memory_request(resources["memory"])
+
+
+class SquashedPipelineGenerator(DefaultPipelineGenerator):
+    def generate_pipeline(self, pipeline, image, image_pull_policy, token):
+        @dsl.pipeline(
+            name=self.get_pipeline_name(),
+            description=self.run_config.description,
+        )
+        def convert_kedro_pipeline_to_kfp() -> None:
+            kfp_op = self._build_kfp_pipeline_op(image, pipeline, token)
+            kfp_op.container.set_image_pull_policy(image_pull_policy)
+
+        return convert_kedro_pipeline_to_kfp
+
+    def _build_kfp_pipeline_op(
+        self,
+        image,
+        pipeline,
+        tracking_token=None,
+    ) -> dsl.ContainerOp:
+        inputs = []
+        command = []
+        env = self.run_config.environment
+        should_add_params = len(self.context.params) > 0
+        name = clean_name(pipeline)
+
+        def _decorate_command(cmd, env):
+            return (
+                " ".join(f"{key}={value}" for key, value in env.items())
+                + " "
+                + cmd
+            )
+
+        if self.run_config.network.host_aliases:
+            command.append(self._generate_hosts_file())
+
+        if should_add_params:
+            command.append(self._generate_params_command(should_add_params))
+
+        if is_mlflow_enabled():
+            inputs.append(InputSpec("mlflow_tracking_token", "String"))
+            env[
+                "MLFLOW_TRACKING_TOKEN"
+            ] = "{{$.inputs.parameters['mlflow_tracking_token']}}"
+            command.append(
+                _decorate_command(
+                    f"kedro vertexai -e {self.context.env} mlflow-start --output /tmp/mlflow.run.id {self.run_name} &&",
```

Comment:
Do we still need this if we're running in a single node?

Reply:
Good question! It has one nice feature of setting the tag with vertex run id, so I decided to keep it. It could be covered by some custom hook in the project, but it requires custom development.
![image](https://user-images.githubusercontent.com/4659069/162448607-a6cab09f-e76a-4e35-8aad-12ca40634d50.png)